src/components/Accounts/ApprovedAccounts.jsx

```diff
@@ -1,67 +1,175 @@
 import { NPOBackend } from '../../utils/auth_utils.js';
-import { useEffect, useState } from 'react';
+import { useEffect, useState, useCallback } from 'react';
 import { Box, Table, Thead, Tbody, Tr, Th, Td, TableContainer, Button, Checkbox, useDisclosure } from '@chakra-ui/react'
 import { CloseIcon } from '@chakra-ui/icons'
 import DeleteAccountModal from './DeleteAccountModal.jsx';
 import PropTypes from 'prop-types';
+import PaginationFooter from "../../components/Catalog/PaginationFooter/PaginationFooter";
+import { usePagination } from '@ajna/pagination';
 
-const ApprovedAccounts = ( {accountType} ) => {
+const ApprovedAccounts = ( {accountType, searchQuery} ) => {
     const [approvedAccounts, setApprovedAccounts] = useState([]);
     const { isOpen: isDeleteOpen, onOpen: onDeleteOpen, onClose: onDeleteClose } = useDisclosure();
     const [deleteItemId, setDeleteItemId] = useState("");
+    const [totalRowCount, setTotalRowCount] = useState(0);
+    const { currentPage, setCurrentPage, pagesCount, offset, pageSize, setPageSize } = usePagination({
+        initialState: { currentPage: 1, pageSize: 10 },
+        pagesCount: Math.ceil(totalRowCount / 10),
+      });
+    const [individualChecked, setIndividualChecked] = useState(new Array(approvedAccounts.length).fill(false));
+    const [checkedAccountIds, setCheckedAccountIds] = useState([]);
+    const [dataShouldRevalidate, setDataShouldRevalidate] = useState(false);
+
+    // useEffect(() => {
+    //     const renderTable = async () => {
+            // try {
+            //     const { data } = await NPOBackend.get(`/users/approved-accounts`, {
+            //         params: {
+            //             keyword: (searchQuery && searchQuery.length) && searchQuery, 
+            //             page: currentPage, 
+            //             limit: pageSize, 
+            //             accountType: accountType}
+            //     });
+            //     setApprovedAccounts(data.accounts);
+            //     setTotalRowCount(Number(data.count[0].count));
+            // } catch (error) {
+            //     console.error('Error fetching data:', error);
+            // }
+    //     };
+    //     renderTable();
+    // }, [searchQuery, totalRowCount, currentPage, pageSize, accountType]);
+
+    const fetchTableData = useCallback(async () => {
+        try {
+            const { data } = await NPOBackend.get(`/users/approved-accounts`, {
+                params: {
+                    keyword: (searchQuery && searchQuery.length) && searchQuery, 
+                    page: currentPage, 
+                    limit: pageSize, 
+                    accountType: accountType}
+            });
+            setApprovedAccounts(data.accounts);
+            setTotalRowCount(Number(data.count[0].count));
+        } catch (error) {
+            console.error('Error fetching data:', error);
+        }
+    }, [searchQuery, totalRowCount, currentPage, pageSize, accountType]);
```

Comment:
⚠️ **[eslint]** <[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>React Hook useCallback has an unnecessary dependency: 'totalRowCount'. Either exclude it or remove the dependency array.